lib/client-assets.php

```diff
@@ -778,6 +780,39 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
 		wp_die( $post_to_edit->get_error_message() );
 	}
 
+	// Add autosave data if it is newer and changed.
+	$autosave      = wp_get_post_autosave( $post->ID );
+	$show_autosave = false;
+
+	// Is the autosave newer than the post?
+	if (
```

Comment:
This logic is slightly more complex than I'm comfortable inlining in the script enqueue function. Could be externalized, ideally with some more comments clarifying what's being done in looping over revision fields.

Reply:
Ok, I can change this. Please note this code comes verbatim from core - `wp-admin/edit-form-advanced.php`

Reply:
extracted this